lib/runner/extensions/request.command.js

```diff
@@ -29,43 +30,46 @@ var _ = require('lodash'),
      * @returns {Object}
      */
     createContext = function (payload) {
-        var context = payload.context ? payload.context : _.pick(payload, CONTEXT_PROPERTIES),
+        var context = _.pick(payload, CONTEXT_PROPERTIES),
+            item,
             parent;
 
-        // get a reference to the Auth instance from the original item, so any changes don't have to be synced back
-        context.auth = payload.item.getAuth();
+        // we clone item from the payload, so that we can make any changes we need there, without mutating the 
+        // collection.
+        item = new sdk.Item(payload.item.toJSON());
 
-        // if an item is not provided in the context, clone the one from the payload,
-        // so that we can make any changes we need there, without mutating the collection.
-        context.item = context.item || new sdk.Item(payload.item.toJSON());
+        // in order to ensure that variable resolution works correctly, we set the __parent property,
+        // so that the SDK can locate the VariableList in the collection.
+        parent = payload.item.parent();
+        parent && (item.setParent(parent));
+
+        // save the cloned item in context
+        context.item = item;
 
+        // save original item for reference
+        context.originalItem = payload.item;
+
+        // get a reference to the Auth instance from the item, so any changes don't have to be synced back
+        !context.auth && (context.auth = context.item.getAuth());
```

Comment:
@shamasis for the case of folder level auth,
- original item's parent will have the auth
- after we clone the auth and set the parent, it's done using reference 
- getAuth also returns a reference
- so now if we change the context.auth, wouldn't it change the original item's parent's auth also?

Reply:
@harryi3t updated code to resolve in `presend` helpers. When variables are resolved, auth is cloned and updating the auth from then on will not update the original item.

Of course, there is a possibility of mutating auth at the time between creating context and resolving variables in `presend`